source/main/GameContext.cpp

```diff
@@ -736,12 +736,20 @@ void GameContext::UpdateGlobalInputEvents()
             }
             else if (App::sim_state->GetEnum<SimState>() == SimState::RUNNING)
             {
-                this->PushMessage(Message(MSG_SIM_PAUSE_REQUESTED));
+                if (App::mp_state->GetEnum<MpState>() != MpState::CONNECTED)
+                {
+                    this->PushMessage(Message(MSG_SIM_PAUSE_REQUESTED));
+                }
+                this->PushMessage(Message(MSG_GUI_OPEN_MENU_REQUESTED));
             }
             else if (App::sim_state->GetEnum<SimState>() == SimState::PAUSED)
             {
                 this->PushMessage(Message(MSG_SIM_UNPAUSE_REQUESTED));
             }
+            if (App::GetGuiManager()->IsVisible_GameMainMenu())
```

Comment:
this check is redundant - you should put the GUI_CLOSE_MENU inside the `if PAUSED` condition above. 

you may also wrap the MSG_UNPAUSE in `!CONNECTED` check - that way, it will also be nicely consistent with the `if RUNNING` condition above.

Reply:
That way there times that Esc does nothing, the only way i managed getting it to work every time is independent from the those checks and after them:

https://github.com/RigsOfRods/rigs-of-rods/blob/6015013855f977950d2e53ed04e8c766f90bf841/source/main/GameContext.cpp#L749-L756,